worlds/messenger/Options.py

```diff
@@ -7,9 +7,30 @@ class MessengerAccessibility(Accessibility):
     __doc__ = Accessibility.__doc__.replace(f"default {Accessibility.default}", f"default {default}")
 
 
-class Logic(DefaultOnToggle):
-    """Whether the seed should be guaranteed completable."""
-    display_name = "Use Logic"
+class Logic(Choice):
+    """
+    The level of logic to use when determining what locations in your world are accessible.
+    Normal can require damage boosts, but otherwise approachable for someone who's beaten the game.
+    Hard has some more difficult tricks in logic, but should be easier for speedrunners.
+    Challenging contains more medium and hard difficulty speedrunning tricks.
+    None places everything with the minimum amount of rules possible. Expect to do OoB.
```

Comment:
Re: "Medium in challenging" it's because the medium difficulty tricks are difficult enough to warrant separating them out of the hard category, but there were so few differences between it and the level above it, it just made more sense to combine them into one option.